### PR TITLE
HMAC-SHA-3

### PR DESCRIPTION
**Pull request by ejohnstown**
Added support for HMAC with SHA-3.

### CODE REVIEW

File: wolfcrypt/src/hmac.c [hunk 1]
```diff
@@ -46,7 +46,7 @@
     {
         if (hmac == NULL || (key == NULL && keySz != 0) ||
            !(type == WC_MD5 || type == WC_SHA || type == WC_SHA256 ||
-                type == WC_SHA384 || type == WC_SHA512 || type == BLAKE2B_ID)) {
+                type == WC_SHA384 || type == WC_SHA512)) {
```
Comment (SparkiDev) [inline wolfcrypt/src/hmac.c]: BLACK2B_ID is removed but SHA3 not added and below BLAKE2B_ID is still being tested.

Comment (ejohnstown) [reply wolfcrypt/src/hmac.c]: I put blake2b back in and fixed the SHA-3 includes/etc to follow the patterns for the inclusion of Blake2B and SHA-224. I also fixed the issue with the MCAPI build and building without SHA-512.

File: wolfcrypt/src/hmac.c [hunk 2]
```diff
@@ -46,7 +46,8 @@
     {
         if (hmac == NULL || (key == NULL && keySz != 0) ||
            !(type == WC_MD5 || type == WC_SHA || type == WC_SHA256 ||
-                type == WC_SHA384 || type == WC_SHA512 || type == BLAKE2B_ID)) {
+                type == WC_SHA384 || type == WC_SHA512 ||
```
Comment (SparkiDev) [inline wolfcrypt/src/hmac.c]: My reading of this code is that you can't set the HMAC key with SHA-3 algorithm type in FIPS.
Is this correct? Are we validating HMAC-SHA-3?

Comment (ejohnstown) [reply wolfcrypt/src/hmac.c]: We are not adding SHA-3 to the existing FIPS boundary. We are adding SHA-3 and HMAC-SHA-3 to the new FIPS boundary for that is a WIP. This PR is for non-FIPS HMAC-SHA-3.

**Pull request merged**